### PR TITLE
ROX-18937: Fix metrics exposure for Compliance and NodeInv. 

### PR DESCRIPTION
**Pull request by vikin91**
## Description

Ensure that Compliance Prometheus server exposes the metrics over HTTP and our monitoring setup can consume them. While working on #7224, I realized that metrics are not being collected - this PR should fix that.

That required to move the Prom server from node scanner to compliance main function to ensure that metrics are collected also for clusters with disabled node scanning.

## Checklist
- [x] Investigated and inspected CI test results

### N/A
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

- [x] CI
- [x] Manual test - deployment on OpenShift with monitoring and looking at the metrics in Prometheus UI

Demonstrating the metric added in #7224  

Screenshot 2023-08-08 at 13 41 27

### CODE REVIEW

File: compliance/collection/compliance/node_scanner.go [hunk 1]
```diff
@@ -36,10 +35,6 @@ func (n *NodeInventoryComponentScanner) Connect(address string) {
 	if !env.NodeInventoryContainerEnabled.BooleanSetting() {
 		log.Infof("Compliance will not call the node-inventory container, because this is not Openshift 4 cluster")
 	} else if env.RHCOSNodeScanning.BooleanSetting() {
-		// Start the prometheus metrics server
-		metrics.NewServer(metrics.NodeInventorySubsystem, metrics.NewTLSConfigurerFromEnv()).RunForever()
-		metrics.GatherThrottleMetricsForever(metrics.NodeInventorySubsystem.String())
-
```
Comment (vikin91) [inline compliance/collection/compliance/node_scanner.go:42]: @stehessel This was a duplicate Prom server that was causing the issues. I thought this code was in Scanner container 🤦

File: pkg/metrics/subsystems.go [hunk 2]
```diff
@@ -5,11 +5,10 @@ type Subsystem string
 
 // These consts enumerate all the subsystems that expose Prometheus metrics.
 const (
-	CentralSubsystem       Subsystem = "central"
-	SensorSubsystem        Subsystem = "sensor"
-	ComplianceSubsystem    Subsystem = "compliance"
-	NodeInventorySubsystem Subsystem = "nodeinventory"
```
Comment (vikin91) [inline pkg/metrics/subsystems.go:11]: This is not required anymore as Node Inventory container does not produce metrics on its own.

Comment (vikin91) [reply pkg/metrics/subsystems.go:11]: I added that in the previous PR (#7224) and this was never used (even in dev), so no breaking change :) Thanks for checking the details!

**Review by jschnath — APPROVED** [review]
Your comments made clear what's going on, CI looks good too. LGTM

**Pull request merged**